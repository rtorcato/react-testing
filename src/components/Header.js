import React from 'react';

const Header = ({ text }) => {
  return (
    <header data-testid="hitag" className="fancy-h1">
      <h1>{text}</h1>
    </header>
  );
};

export default Header;
