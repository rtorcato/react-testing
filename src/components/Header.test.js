import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Header from './Header';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<Header text="Hello" />);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts text into h1', () => {
  const { getByTestId, getByText } = render(<Header text="Hello!" />);
  expect(getByTestId('hitag')).toHaveTextContent('Hello!');
  expect(getByText('Hello!')).toHaveClass('fancy-h1');
});
