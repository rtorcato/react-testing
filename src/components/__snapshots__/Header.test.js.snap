// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header
    class="fancy-h1"
    data-testid="hitag"
  >
    <h1>
      Hello
    </h1>
  </header>
</DocumentFragment>
`;
